src/command/mcl.zig

```diff
@@ -624,11 +628,51 @@ pub fn init(c: Config) !void {
         .execute = &mclSliderStopPull,
     });
     errdefer _ = command.registry.orderedRemove("STOP_PULL_SLIDER");
+    try command.registry.put("SET_LOG_REGISTER", .{
+        .name = "SET_LOG_REGISTER",
+        .parameters = &[_]command.Command.Parameter{
+            .{ .name = "register list" },
+            .{ .name = "path", .optional = true },
+        },
+        .short_description = "Setup for LOG_REGISTER command.",
+        .long_description =
+        \\Create a log file for logging register value based on the register list
+        \\provided by the user. The "register list" shall be written with comma "," 
+        \\separator, e.g. X,Wr to log register X and Wr only. Valid registers are 
+        \\X, Y, Wr, and Ww. If a path is not provided, the the default logging file
+        \\containing all the register value triggerred by LOG_REGISTER will be 
+        \\created in the current working directory as the following:
+        \\"mmc-register-[register list]-YYYY.MM.DD-HH.MM.SS.csv". 
+        \\
+        \\Note that this command will not log any register value, the register 
+        \\will be logged by LOG_REGISTER command. 
+        ,
+        .execute = &setLogRegister,
+    });
+    errdefer _ = command.registry.orderedRemove("SET_LOG_REGISTER");
+    try command.registry.put("LOG_REGISTER", .{
+        .name = "LOG_REGISTER",
+        .parameters = &[_]command.Command.Parameter{
+            .{ .name = "line name" },
+            .{ .name = "axis" },
+        },
+        .short_description = "Log the specified register value.",
+        .long_description =
+        \\Poll the specified register and write the value to logging file created
+        \\by SET_LOG_REGISTER. The station register value to be saved depends on the
+        \\provided axis.
+        ,
+        .execute = &logRegister,
+    });
+    errdefer _ = command.registry.orderedRemove("LOG_REGISTER");
 }
 
 pub fn deinit() void {
     arena.deinit();
     line_names = undefined;
+    if (register_log_file) |f| {
+        f.close();
+    }
```

Comment:
Not sure if it is required. Please review.

Reply:
This is indeed required, but I think in addition to closing the file it should also set `register_log_file` to `null`.